scapy/utils6.py

```diff
@@ -59,12 +60,14 @@ def cset_sort(x, y):
         return -res
 
     cset = []
-    if in6_isgladdr(addr) or in6_isuladdr(addr):
```

Comment:
I am not convinced that this change is correct. Did you test it on Linux with unique local addresses? It seems that the Linux kernel does not specify IPV6_ADDR_UNIQUE_LOCAL. Therefore, this patch will likely break address selection.